hr_expense_product_config/models/hr_expense.py

```diff
@@ -0,0 +1,53 @@
+# © 2025 Solvos Consultoría Informática (<http://www.solvos.es>)
+# License LGPL-3 - See https://www.gnu.org/licenses/lgpl-3.0.html
+
+from odoo import _, api, models,fields
+from odoo.exceptions import ValidationError
+
+
+class HrExpense(models.Model):
+    _inherit = "hr.expense"
+
+    payment_locked = fields.Boolean(compute="_compute_payment_locked")
+    allowed_product_ids = fields.Many2many(
+        'product.product',
+        compute='_compute_allowed_product_ids'
+    )
+
+    @api.depends('employee_id', 'company_id')
+    def _compute_allowed_product_ids(self):
+        for record in self:
+            config_ids = self.env['hr.employee.expense.config'].search([
+                ('expense_employee_id', '=', record.employee_id.id),
+                '|',
+                ('company_id', '=', record.company_id.id),
+                ('company_id', '=', False),
+            ])
```

Comment:
Could you try this (faster)?
```suggestion
                ('company_id', 'in', [False, record.company_id.id]),
            ])
```